upload_docs.sh is busted

http://build.servo.org/builders/doc/builds/1220/steps/shell/logs/stdio
```
make: Entering directory `/home/servo/buildbot/slave/doc/build/components/script'
make: makefile.cargo: No such file or directory
make: *** No rule to make target `makefile.cargo'.  Stop.
make: Leaving directory `/home/servo/buildbot/slave/doc/build/components/script'
```

CC @vvuk

--- a/components/script/CMakeLists.txt
+++ b/components/script/CMakeLists.txt
@@ -72,6 +72,22 @@ add_custom_command(
   VERBATIM
   )
 
+add_custom_command(
+  OUTPUT apis.html
+  COMMAND python -B ${bindings_src}/pythonpath.py -I ${bindings_src}/parser -I ${bindings_src}/ply
+                 ${bindings_src}/GlobalGen.py
+                 --cachedir=_cache
+                 --filelist=webidls.list
+                 --only-html
+                 ${bindings_src}/Bindings.conf
+                 .
+                 ${PROJECT_SOURCE_DIR}
+  DEPENDS _cache ${globalgen_deps} ${webidls}
+  VERBATIM
+  )
+
+add_custom_target(supported-apis DEPENDS apis.html)
+
 # We need an intermediate custom target for this, due to this misfeature:
 # > If any dependency is an OUTPUT of another custom command in the same
 # > directory CMake automatically brings the other custom command into the

--- a/etc/ci/upload_docs.sh
+++ b/etc/ci/upload_docs.sh
@@ -20,9 +20,11 @@ cp etc/doc.servo.org/* target/doc/
 
 python components/style/properties/build.py servo html
 
-OUT_DIR="$(pwd)/target/doc/servo" \
-  make -f makefile.cargo -C components/script dom_docs
-rm -rf target/doc/servo/.cache
+cd components/script
+cmake .
+cmake --build . --target supported-apis
+cp apis.html target/doc/servo/
+cd ../..
 
 ghp-import -n target/doc
 git push -qf "https://${TOKEN}@github.com/servo/doc.servo.org.git" gh-pages

